pkg/proxy/iptables/proxier.go

```diff
@@ -572,45 +590,56 @@ func (proxier *Proxier) OnEndpointsSynced() {
 	proxier.syncProxyRules(syncReasonEndpoints)
 }
 
-// Convert a slice of api.Endpoints objects into a map of service-port -> endpoints.
-func buildNewEndpointsMap(allEndpoints endpointsMap, curMap proxyEndpointMap, hostname string) (newMap proxyEndpointMap, hcEndpoints map[types.NamespacedName]int, staleSet map[endpointServicePair]bool) {
+func (proxier *Proxier) mergeEndpointsMap(endpointsMap proxyEndpointsMap) {
+	for svcPort := range endpointsMap {
+		proxier.endpointsMap[svcPort] = endpointsMap[svcPort]
+	}
 
-	// return values
-	newMap = make(proxyEndpointMap)
-	hcEndpoints = make(map[types.NamespacedName]int)
-	staleSet = make(map[endpointServicePair]bool)
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ExternalTrafficLocalOnly) {
```

Comment:
I think we can do it (at least for now). If it become a problem, we can get back to it.
Will change.